lib/facility.rb

```diff
@@ -1,14 +1,78 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
+  def initialize(facilities_info)
+    @name = facilities_info[:name]
+    @address = facilities_info[:address]
+    @phone = facilities_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+  
   end
-
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    # return false unless @services.include?('Register Vehicle')
```

Comment:
yes, we do need to check first if the services include that string, otherwise we don't want to do any of this logic. You could have on line 17: `if @services.include?("Register Vehicle")` and then put all the logic inside that block.